README.md
# Siritie

Siritie is a challenging survival game that will put your skills to the test as
you fight to survive in a harsh and mystical world. Created by Ozcodex, Siritie
is a derivation of the original game _Exile v0.3.9e_ by Dokimi, with
contributions from Mantar. The game is designed for `Minetest 5.3+` and offers
an immersive and exciting gameplay experience.

## Installation

To get started with Siritie, simply clone the repository into your
`minetest/games` folder and you're good to go!

## Gameplay

In Siritie, you'll find yourself in a wilderness survival scenario with limited resources, simple technology, and dynamic weather conditions. You'll need to rely on your wits to find food, water, and shelter, while exploring the mysterious world and developing your skills to endure your exile. You'll need to be careful, however, as the world is full of dangers, and one wrong move could spell the end of your adventure.

### Features

*Player physical conditions* - Battle against the elements, the hunger and 
various illnesses, including hypothermia, exhaustion, and disease, as you strive
to survive.
*Dynamic living world* - Immerse yourself in a world with ever-changing seasonal
weather patterns, erosion, and water flows through soil, adding to the realism
and challenge.
*Realistic building materials* - Build your own shelter from the rain, create
kilns and smelters, and use realistic and sustainable materials to craft your
tools and equipment.
*Authentic crafting* - Gather natural resources and materials to create tools,
clothing, and food that are crucial for your survival, making your experience
more immersive.
*A Hint of Mysticism* -  Explore the mystical realm of alchemy, and seek
blessings by offering prayers to the gods. It's uncertain whether these gods
truly exist, or if the magical elements even hold any real power.
*Unique Characters* - Each character in Siritie is unique, born with a
backstory, name, and land of origin, with distinctive characteristics that shape
their journey through the game.

### Tips and Tricks

As you explore and survive in Siritie, you will discover that there are many different strategies that may work, and part of the fun is figuring out what does, and catastrophically does not, work. Here are some early advices you might pass through to ensure your survival:

The first step is to find a comfortable shelter that will help regulate your body temperature and protect you from hypothermia or heat stroke.

Once you have a shelter, it's important to conserve your energy by doing tasks such as crafting, organizing, and planning while you rest. Extreme weather can sap your energy quickly, so avoid walking around in snowstorms or other dangerous weather.

If you're in a situation where you need to build a shelter, it's essential to have a fireplace or plenty of torches to regulate temperature. Consider going underground in caves that are deep enough to be safe from the weather. Collect water in pots from cave drips or rainwater, and seek out plants and food that quench your thirst. As a last resort, you can try to melt ice or dig a pit in wet ground and wait for water to flow into it.

To sustain yourself, you can catch animals with clubs and explore farming for a sustainable food source. Simple grass clothing can be woven quickly and can be a lifesaver, while better clothes can be made by soaking bundles of the right kind of plants. Keep an eye on "health effects," as you may have eaten something bad or have a terrible disease. Some plants have useful medicinal effects, but be careful not to overdose.

Fibrous plants, sticks and stones are fine as initial tools, but woodcutting requires better tools for harder woods. Build stairs and shelters around your base to save energy and protect you from extremes. Use sticks to build a ladder or pole to shimmy up, to get to high places or descend carefully.

Not every step in crafting can be done at a work station, so consider the environment and industry you need to create to advance. Build ovens, kilns, and furnaces just as you would in real life, and start a fire with access to air and a sealed chamber that gets heated up. Blocks are hotter than slabs, and charcoal is hotter than wood. Two slabs, one on top of another chreates a whole block. Fires can be temporarily extinguished by punching while holding sediment, and charcoal can be made by creating a wood fire sealed up to limit airflow.

If you're looking to make green or clear glass, sand and wood-ash can be made into green glass, but clear glass requires the ash to be soaked, dried, and roasted. Glass can be melted onto iron trays to make panes for real windows! Iron smelting is hard and requires plenty of charcoal and a space below the iron mixture for slag to drain out.

Overall, surviving in extreme weather conditions is all about being prepared and knowing what to do. With the right knowledge, anyone can make it through even the toughest situations.

## World Settings

Valleys is the standard `mapgen` for _Siritie_. `Carpathian` is also supported,
for a somewhat more difficult and slower-paced game.

## Settings for Multiplayer
The variable `time_speed` defaults to `72`, and at this rate a player who logs
on at 8:00 AM every day will get a change in season every 2.5 days.

Changing speed to `60` will make days last 24 minutes and a new season every 
real-world day, but he will see the seasons in reverse. At `time_speed` of `96`,
days last only 15 minutes, but the player might see spring (year 1) on day 1,
summer (year 2) on day 2, etc.

Set `exile_hud_update` to `1.0` second for multiplayer servers on the Internet;
a LAN server can probably handle `0.2` seconds.

## Mods for Multiplayer
- [Wield3d](https://github.com/stujones11/wield3d) is recommended.

## Development
_Siritie_ is open-source software - that means the game is as good as you choose
to make it. It also means development can be erratic and haphazard at times, so be patient!

_Siritie_ is currently in “Alpha,” therefore you can expect that there may be bugs, missing features, performance issues, and perhaps compatibility-breaking updates.
Despite this, _Siritie_ does have enough features to be a playable game and should be stable and mostly bug-free.

## Credits
I would like to express our heartfelt gratitude to Dokimi for creating the original game Exile, which served as the foundation for Siritie's development. Also, thanks to Mantar for updating and contributing to the original game, which has indirectly contributed to the development of Siritie. 

Gratitude is due to all those whose mods have been adapted for use in _Siritie_ (see `./mod/` folders for details).

Thanks also to all who have given feedback, fixes, etc. 

## License

Copyright (C) 2019-2023 Dokimi, Mantar and OzCodex

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <https://www.gnu.org/licenses/>.

CHANGELOG.md
# Changelog:

## [2023-05-26] 0.4.1
  - Code Refactor
  - Expand traits in character Tab
  - Introduce specialized roof ridge tiles
  - Utilize digging stick for path creation instead of agricultural soil
  - Enhance alembic with alcohol extraction capability
  - Implement Wiha fermentation process for sugar and vinegar production
  - Eliminate slag stairs and slabs due to their lack of usefulness
  - Remove boulders from built blocks since soome actions are irreversible
  - Replace mixing spot with craftable crafts (mixing bucket and masonry toolset)
  - Introduce new craft station: wattlework station
  - Organize tech and textures folder with consistent naming convention
  - Enable composting of meats and cooked food and change composter output from loam to silt
  - Utilize groups for define the amount of compost generated by the composter
  - Introduce clay mixing bucket to undo unfired pottery
  - Enable oil extraction from maraka nut
  - Introduce pottery wheel as an improved clay shaping spot
  - Offer alternative hammering log made from granite
  - Introduce fishtrap tool to automatically catch fishes in the sea
  - Implement clay smoker for bulk cooking
  - Add amphora for tang and wiha cider
  - Implement ceramic pendulum seismograph inspired by Zhang Heng's Seismoscope
  - Introduce alchemy ceramic equipment: Alchemy Mortarium, Elemental Crucible
  - Add alchemy glass equipment: Retort and Vessel
  - Create alchemy bench from alchemy equipement and add alchemy recipes
  - Introduce aluminium as material
  - Add aluminium windows and water containers

## [2023-05-10] 0.4
  - Ozcodex's restructuration
  - Add idol worshipping, with blessings, curses and traits
  - New working spot for all the pottery
  - Implement a composter block to turn organics into fertilizer
  - Add Ceramic Watering Can to water blocks and Alembic to extract salt from water
  - Turn bones into bonemeal on the grinding stone
  - Add charcoal powder to mortar and pestle
  - Implements black and red painting
  - Add sundial to get info about the date
  - Adds a great rebalance of all recipes
  - Bring back the weaving frame, the chopping block and the hammering log
  - Adds Mixing Bucket and Mortar Bucket
  - Remove rings and hammers
  - Remove grinding, hammering and weaving spots
  - Remove crafting recipes on workspaces that creates themselves
  - Remove bulk recipes

## [2023-04-07] 0.3.9e
  - Add theme song "Earth and Stone" by Kevin Hartnell
  - Wattle connects to drystack
  - Keep leftover plant bundles that are not dyes, to see what you've tried
  - Added a dewaffler utility to fix broken terrain caused by the prior bug
  - Fix for opaque white fog banks
  - Usual round of bugfixes

## [2022-12-31] 0.3.9d
  - Add an alignment tool to complement the lever
  - Fix grass spreading, grass dying in shade, and landscape waffling
  - Gateway effects for incoming characters, courtesy of Dokimi
  - Better intro screen
  - Bugfixes

## [2022-12-076 0.3.9c
  - Misc bugfixes

## [2022-11-26] 0.3.9b
  - Volcanic ash on volcanoes
  - Bugfixes

## [2022-11-04] 0.3.9
  - New HUD design
  - Iron lantern for mid-late game players
  - iron nails to add protection to nodes
  - bed can now store a blanket
  - cobbles, small loose stones to complement boulders
  - ability to combine small wood fires after burning
  - animal probe can get player stats
  - options for double-click to eat: /eat2x, and wide hud: /hud16 
  - "crafted by" labels for craft stations
  - Folded Minetest 5.3.0 compatibility into mainline
  - More bugfixes than you can shake a digging stick at

## [2022-08-05] 0.3.8d and 0.2.11d
  - Minor bugfixes

## [2022-08-05] 0.3.8c and 0.2.11c
  - enabled shadows when running on MT 5.6.0+
  - Pots fill with rainwater even if the player is away
  - Many, many bug fixes

## [2022-07-08] 0.3.8b and 0.2.11b
  - Cheaper rammed earth
  - Improved Sarkamos life cycle and behavior
  - Cooking pot improvements
  - Fixed wooden & iron chests' trashcan icon, dig & replace to reload
  - Misc bugfixes

## [2022-06-25] 0.3.8a and 0.2.11a
  - Reverted betterfall mod due to problems on some systems

## [2022-06-24] 0.3.8 and 0.2.11
  - New tech rebalance, multiple ways up the tech tree
  - Massive rings left scattered on the surface by the Ancients
  - Throwable spears for hunting and fighting
  - Improved canoe and airboat performance; the airboat strafes with AUX1
  - Fire resistance for tree trunks and other nodes
  - Ability to cut sedimentary rock blocks intact with a chisel
  - Clump falling for drystack/thatch, build ceilings with supports
  - Better, more survivable gundu fish, enables fish farming
  - Can eat unmelted snow for thirst, with drawbacks
  - Spanish translation


## [2022-05-28] 0.3.7 and 0.2.10
  - balance: limited fire spread for multiplayer
  - many airboat improvements, including strafe movement with AUX1
  - Haze overlay warns of extreme heat
  - animal probes to watch the health and age of livestock
  - underwater plants spread and replenish
  - beds and thatch reduce fall damage
  - some valuable nodes have protection in multiplayer
  - beginnings of i18n French translation, early translation support

## [2022-04-16] 0.3.6 and 0.2.9
 - Craftable dyes, made from randomly selected plants
 - Wielded_light replaces Illumination mod
 - Iron chests and basic protection for valuable items
 - Trees regrow after a rainy season, not only during rain
 - Transfer water directly between pots and bottles
 - Further cooking pot improvements

## [2022-02-25] 0.3.5b and 0.2.8b
 - flowing water no longer supports player weight in MT 5.5.0
 - /suicide command for players who are stuck
 - added frost overlay to the HUD when you're dangerously cold
 - cooking pot now available, still somewhat WIP
 - many minor bugfixes and improvements

## [2022-01-21] 0.3.5 and 0.2.8
  - Blankets to keep warm.
  - Foods can now burn if left cooking too long.
  - Mobs should no longer `stun`-lock when hit repeatedly.

mods/tech/pottery.lua
------------------------------------------------
-- POTTERY
-- all made from clay
--unfired versions must reach right temperature for long enough to fire.
-----------------------------------------------

-- Internationalization
local S = tech.S

local c_alpha = minimal.compat_alpha

--firing difficulty
local base_firing = ncrafting.base_firing
local firing_int = ncrafting.firing_int
sediment = sediment
lightsource = lightsource
lightsource_description = lightsource_description

---
--Broken Pottery
--if you smash it up, or from failed firings
--slab
minetest.register_node("tech:broken_pottery", {
	description = S("Broken Pottery"),
	tiles = {"tech_broken_pottery.png"},
	stack_max = minimal.stack_max_bulky *2,
	paramtype = "light",
	drawtype = "nodebox",
	node_box = {
		type = "fixed",
		fixed = {-0.5, -0.5, -0.5, 0.5, 0, 0.5},
	},
	groups = {cracky = 3, falling_node = 1, oddly_breakable_by_hand = 3},
	sounds = nodes_nature.node_sound_gravel_defaults(),
        on_rightclick = function (pos,node,clicker,itemstack,pointed_thing)
           return minimal.slabs_combine(pos,node,itemstack,"tech:broken_pottery_block")
	end,
})

-- Broken pottery full blocks and soil
local broken_pottery =
   sediment.new({name = "broken_pottery_block",
		 description = S("Broken Pottery"),
		 hardness = sediment.hardness.soft,
		 fertility = 5, sound = sediment.sounds.gravel,
		 sound_wet = sediment.sounds.gravel_wet,
		 texture_name = "tech_broken_pottery.png", mod_name = "tech"})
sediment.register_dry(broken_pottery)
sediment.register_wet(broken_pottery)
sediment.register_wet_salty(broken_pottery)
sediment.register_agri_soil_variants(broken_pottery)

-------------------------------------------------------------------
--THIS SHOULD BE MOVED somewhere GENERALIZED to handle non-pottery pots
function water_pot(pos, pot_name, elapsed)
	local light = minimal.get_daylight({x=pos.x, y=pos.y + 1, z=pos.z}, 0.5)
	--collect rain
	if light == 15 then
	   if climate.get_rain(pos, light) or
	      climate.time_since_rain(elapsed) > 0 then
	          minetest.swap_node(pos, {name = pot_name.."_freshwater"})
		  return
	   end
	else
		--drain wet sediment into the pot
		--or melt snow and ice
		local posa = 	{x = pos.x, y = pos.y+1, z = pos.z}
		local name_a = minetest.get_node(posa).name
		if name_a == "air" then
			return true
		--[[-- Water puddles make more sense than this now,
		--especially given disease risks
		elseif minetest.get_item_group(name_a, "wet_sediment") == 1 then
			local nodedef = minetest.registered_nodes[name_a]
			if not nodedef then
				return true
			end
			minetest.set_node(posa, {name = nodedef._dry_name})
			minetest.set_node(pos, {name = "tech:clay_water_pot_freshwater"})
			return
		elseif minetest.get_item_group(name_a, "wet_sediment") == 2 then
			local nodedef = minetest.registered_nodes[name_a]
			if not nodedef then
				return true
			end
			minetest.set_node(posa, {name = nodedef._dry_name})
			minetest.set_node(pos, {name = "tech:clay_water_pot_salt_water"})
			return
			]]
		elseif (name_a == "nodes_nature:ice" or
			name_a == "nodes_nature:snow_block" or
			name_a == "nodes_nature:freshwater_source" ) then
			if climate.can_thaw(posa) then
				minetest.swap_node(pos, {name = pot_name.."_freshwater"})
				minetest.remove_node(posa)
				return
			end
		end
	end
	return true
end




--Water pot
--for collecting water, catching rain water
--fired
minetest.register_node("tech:clay_water_pot", {
	description = S("Clay Water Pot"),
	tiles = {
		"tech_water_pot_empty.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png"
	},
	drawtype = "nodebox",
	stack_max = minimal.stack_max_bulky,
	paramtype = "light",
	node_box = {
		type = "fixed",
		fixed = {
			{-0.25, 0.375, -0.25, 0.25, 0.5, 0.25}, -- NodeBox1
			{-0.375, -0.25, -0.375, 0.375, 0.3125, 0.375}, -- NodeBox2
			{-0.3125, -0.375, -0.3125, 0.3125, -0.25, 0.3125}, -- NodeBox3
			{-0.25, -0.5, -0.25, 0.25, -0.375, 0.25}, -- NodeBox4
			{-0.3125, 0.3125, -0.3125, 0.3125, 0.375, 0.3125}, -- NodeBox5
		}
	},
	liquids_pointable = true,
	on_use = function(itemstack, user, pointed_thing)
		return liquid_store.on_use_empty_bucket(itemstack, user, pointed_thing)
	end,
		--collect rain water
	on_construct = function(pos)
		minetest.get_node_timer(pos):start(math.random(30,60))
	end,
	on_timer =function(pos, elapsed)
		return water_pot(pos, "tech:clay_water_pot", elapsed)
	end,
	groups = {dig_immediate = 3, pottery = 1, temp_pass = 1},
	sounds = nodes_nature.node_sound_stone_defaults(),

})



--unfired
minetest.register_node("tech:clay_water_pot_unfired", {
	description = S("Clay Water Pot (unfired)"),
	tiles = {
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png"
	},
	drawtype = "nodebox",
	stack_max = minimal.stack_max_bulky,
	paramtype = "light",
	node_box = {
		type = "fixed",
		fixed = {
			{-0.25, 0.375, -0.25, 0.25, 0.5, 0.25}, -- NodeBox1
			{-0.375, -0.25, -0.375, 0.375, 0.3125, 0.375}, -- NodeBox2
			{-0.3125, -0.375, -0.3125, 0.3125, -0.25, 0.3125}, -- NodeBox3
			{-0.25, -0.5, -0.25, 0.25, -0.375, 0.25}, -- NodeBox4
			{-0.3125, 0.3125, -0.3125, 0.3125, 0.375, 0.3125}, -- NodeBox5
		}
	},
	groups = {dig_immediate=3, temp_pass = 1, heatable = 20},
	sounds = nodes_nature.node_sound_stone_defaults(),
	on_construct = function(pos)
		--length(i.e. difficulty of firing), interval for checks (speed)
		ncrafting.set_firing(pos, base_firing, firing_int)
	end,
	on_timer = function(pos, elapsed)
		--finished product, length
		return ncrafting.fire_pottery(pos, "tech:clay_water_pot_unfired", "tech:clay_water_pot", base_firing)
	end,
})


--------------------------------------
--unfired storage pot (see storage for fired version)
minetest.register_node("tech:clay_storage_pot_unfired", {
	description = S("Clay Storage Pot (unfired)"),
	tiles = {
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png"
	},
	drawtype = "nodebox",
	stack_max = minimal.stack_max_bulky,
	paramtype = "light",
	node_box = {
		type = "fixed",
		fixed = {
				{-0.375, -0.5, -0.375, 0.375, -0.375, 0.375},
				{-0.375, 0.375, -0.375, 0.375, 0.5, 0.375},
				{-0.4375, -0.375, -0.4375, 0.4375, -0.25, 0.4375},
				{-0.4375, 0.25, -0.4375, 0.4375, 0.375, 0.4375},
				{-0.5, -0.25, -0.5, 0.5, 0.25, 0.5},
			}
		},
	groups = {dig_immediate=3, temp_pass = 1, heatable = 20},
	sounds = nodes_nature.node_sound_stone_defaults(),
	on_construct = function(pos)
		--length(i.e. difficulty of firing), interval for checks (speed)
		ncrafting.set_firing(pos, base_firing+5, firing_int)
	end,
	on_timer = function(pos, elapsed)
		--finished product, length
		return ncrafting.fire_pottery(pos, "tech:clay_storage_pot_unfired", "tech:clay_storage_pot", base_firing+5)
	end,
})

--------------------------------------
--OIL LAMP

--unfired oil clay lamp
minetest.register_node("tech:clay_oil_lamp_unfired", {
	description = S("Clay Oil Lamp (unfired)"),
	tiles = {
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png",
		"nodes_nature_clay.png"
	},
	drawtype = "nodebox",
	stack_max = minimal.stack_max_medium,
	paramtype = "light",
	paramtype2 = "facedir",
	use_texture_alpha = c_alpha.clip,
	node_box = {
		type = "fixed",
		fixed = {
			--{-0.0625, -0.125, 0.25, 0.0625, 0.0625, 0.4375}, -- flame
			{-0.125, -0.5, -0.125, 0.125, -0.4375, 0.125}, -- bottom
			{-0.0625, -0.4375, -0.0625, 0.0625, -0.3125, 0.0625}, -- stand
			{-0.125, -0.3125, -0.125, 0.125, -0.125, 0.125}, -- body
			{-0.0625, -0.25, 0.125, 0.0625, -0.125, 0.25}, -- spout
			{-0.0625, -0.1875, -0.1875, 0.0625, -0.125, -0.125}, -- handle
			{-0.0625, -0.3125, -0.1875, 0.0625, -0.25, -0.125}, -- handle
			{-0.0625, -0.3125, -0.25, 0.0625, -0.125, -0.1875}, -- handle
		}
	},
	groups = {dig_immediate=3, temp_pass = 1, falling_node = 1, heatable = 20},
	sounds = nodes_nature.node_sound_stone_defaults(),
	on_construct = function(pos)
		--length(i.e. difficulty of firing), interval for checks (speed)
		ncrafting.set_firing(pos, base_firing, firing_int)
	end,
	on_timer = function(pos, elapsed)
		--finished product, length
		return ncrafting.fire_pottery(pos, "tech:clay_oil_lamp_unfired", "tech:clay_oil_lamp_empty", base_firing)
	end,
})

local oil_lamp_desc = lightsource_description.new(
    {lit_name = "tech:clay_oil_lamp", unlit_name = "tech:clay_oil_lamp_unlit",
     fuel_name = "tech:vegetable_oil", max_fuel = 3100,
     burn_rate = 5, refill_ratio = 1/2, put_out_by_moisture = true})


--fired oil clay lamp
minetest.register_node("tech:clay_oil_lamp_unlit", {
	description = S("Clay Oil Lamp"),
	tiles = {
		"tech_oil_lamp_top.png",
		"tech_oil_lamp_bottom.png",
		"tech_oil_lamp_side.png",
		"tech_oil_lamp_side2.png",
		"tech_oil_lamp_front.png",
		"tech_oil_lamp_front.png"
	},
	drawtype = "nodebox",
	stack_max = minimal.stack_max_medium,
	paramtype = "light",
	paramtype2 = "facedir",
	use_texture_alpha = c_alpha.clip,
	node_box = {
		type = "fixed",
		fixed = {
			--{-0.0625, -0.125, 0.25, 0.0625, 0.0625, 0.4375}, -- flame
			{-0.125, -0.5, -0.125, 0.125, -0.4375, 0.125}, -- bottom
			{-0.0625, -0.4375, -0.0625, 0.0625, -0.3125, 0.0625}, -- stand
			{-0.125, -0.3125, -0.125, 0.125, -0.125, 0.125}, -- body
			{-0.0625, -0.25, 0.125, 0.0625, -0.125, 0.25}, -- spout
			{-0.0625, -0.1875, -0.1875, 0.0625, -0.125, -0.125}, -- handle
			{-0.0625, -0.3125, -0.1875, 0.0625, -0.25, -0.125}, -- handle
			{-0.0625, -0.3125, -0.25, 0.0625, -0.125, -0.1875}, -- handle
		}
	},
	groups = {dig_immediate=3, pottery = 1, temp_pass = 1, falling_node = 1},
	sounds = nodes_nature.node_sound_stone_defaults(),
	floodable = true,
	on_flood = function(pos, oldnode, newnode)
            minetest.add_item(pos, ItemStack("tech:clay_oil_lamp_unlit 1"))
            return false
	end,
        on_construct = function(pos)
            lightsource.update_fuel_infotext(oil_lamp_desc, pos)
        end,
        after_place_node = function(pos, placer, itemstack, pointed_thing)
            lightsource.restore_from_inventory(oil_lamp_desc, pos, itemstack)
            lightsource.update_fuel_infotext(oil_lamp_desc, pos)
        end,
        on_dig = function(pos, node, digger)
            lightsource.save_to_inventory(oil_lamp_desc, pos, digger, false)
        end,
        on_ignite = function(pos, user)
            lightsource.ignite(oil_lamp_desc, pos)
        end,
        on_rightclick = function(pos, node, clicker, itemstack, pointed_thing)
            lightsource.refill(oil_lamp_desc, pos, clicker, itemstack)
        end,
})

--full oil clay lamp
minetest.register_node("tech:clay_oil_lamp", {
	description = S("Clay Oil Lamp (lit)"),
	tiles = {
		"tech_oil_lamp_top.png",
		"tech_oil_lamp_bottom.png",
		{
                    name = "tech_oil_lamp_side_animated.png",
                    animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 3.3}
		},
		{
                    name = "tech_oil_lamp_side2_animated.png",
                    animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 3.3}
		},
		"tech_oil_lamp_front.png",
		"tech_oil_lamp_front.png"
	},
	drawtype = "nodebox",
	stack_max = minimal.stack_max_medium,
	sunlight_propagates = true,
	light_source = 7,
	use_texture_alpha = c_alpha.clip,
	paramtype = "light",
	paramtype2 = "facedir",
	node_box = {
		type = "fixed",
		fixed = {
			{-0.0625, -0.125, 0.25, 0.0625, 0.0625, 0.4375}, -- flame
			{-0.125, -0.5, -0.125, 0.125, -0.4375, 0.125}, -- bottom
			{-0.0625, -0.4375, -0.0625, 0.0625, -0.3125, 0.0625}, -- stand
			{-0.125, -0.3125, -0.125, 0.125, -0.125, 0.125}, -- body
			{-0.0625, -0.25, 0.125, 0.0625, -0.125, 0.25}, -- spout
			{-0.0625, -0.1875, -0.1875, 0.0625, -0.125, -0.125}, -- handle
			{-0.0625, -0.3125, -0.1875, 0.0625, -0.25, -0.125}, -- handle
			{-0.0625, -0.3125, -0.25, 0.0625, -0.125, -0.1875}, -- handle
		}
	},
	groups = {dig_immediate=3, pottery = 1, temp_pass = 1, falling_node = 1, not_in_creative_inventory = 1},
	sounds = nodes_nature.node_sound_stone_defaults(),
	floodable = true,
	on_flood = function(pos, oldnode, newnode)
            minetest.add_item(pos, ItemStack("tech:clay_oil_lamp_unlit 1"))
            return false
	end,
        on_construct = function(pos)
            lightsource.start_burning(oil_lamp_desc, pos)
	end,
        on_timer = function(pos, elapsed)
            return lightsource.burn_fuel(oil_lamp_desc, pos)
	end,
        after_place_node = function(pos, placer, itemstack, pointed_thing)
            lightsource.restore_from_inventory(oil_lamp_desc, pos, itemstack)
        end,
        on_dig = function(pos, node, digger)
            lightsource.save_to_inventory(oil_lamp_desc, pos, digger, false)
        end,
        on_rightclick = function(pos, node, clicker, itemstack, pointed_thing)
	   lightsource.extinguish(oil_lamp_desc, pos)
	   lightsource.update_fuel_infotext(oil_lamp_desc, pos)
        end,
})

---------------------------------------
--Recipes

--
--Hand crafts (clay shaping spot)
--

--Pot from clay
crafting.register_recipe({
	type = "clay_shaping_spot",
	output = "tech:clay_water_pot_unfired 1",
	items = {"nodes_nature:clay_wet 4"},
	level = 1,
	always_known = true,
})

crafting.register_recipe({
	type = "mixing_spot",
	output = "nodes_nature:clay 4",
	items = {"tech:clay_water_pot_unfired 1"},
	level = 1,
	always_known = true,
})


--storage Pot from clay
crafting.register_recipe({
	type = "clay_shaping_spot",
	output = "tech:clay_storage_pot_unfired 1",
	items = {"nodes_nature:clay_wet 6"},
	level = 1,
	always_known = true,
})

crafting.register_recipe({
	type = "mixing_spot",
	output = "nodes_nature:clay 6",
	items = {"tech:clay_storage_pot_unfired 1"},
	level = 1,
	always_known = true,
})

--oil lamp
crafting.register_recipe({
	type = "clay_shaping_spot",
	output = "tech:clay_oil_lamp_unfired 1",
	items = {"nodes_nature:clay_wet"},
	level = 1,
	always_known = true,
})

crafting.register_recipe({
	type = "mixing_spot",
	output = "nodes_nature:clay",
	items = {"tech:clay_oil_lamp_unfired 1"},
	level = 1,
	always_known = true,
})

--Break up pots
crafting.register_recipe({
	type = "mixing_spot",
	output = "tech:broken_pottery",
	items = {"group:pottery"},
	level = 1,
	always_known = true,
})

--Combine broken pottery slabs and vice versa
crafting.register_recipe({
	type = "mixing_spot",
	output = "tech:broken_pottery_block",
	items = {"tech:broken_pottery 2"},
	level = 1,
	always_known = true,
})

crafting.register_recipe({
	type = "mixing_spot",
	output = "tech:broken_pottery 2",
	items = {"tech:broken_pottery_block"},
	level = 1,
	always_known = true,
})


-----------------------------------------------
--Register water stores
--source, nodename, nodename_empty, tiles, node_box, desc, groups

--clay pot with salt water
liquid_store.register_stored_liquid(
	"nodes_nature:salt_water_source",
	"tech:clay_water_pot_salt_water",
	"tech:clay_water_pot",
	{
		"tech_water_pot_water.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png"
	},
	{
		type = "fixed",
		fixed = {
			{-0.25, 0.375, -0.25, 0.25, 0.5, 0.25}, -- NodeBox1
			{-0.375, -0.25, -0.375, 0.375, 0.3125, 0.375}, -- NodeBox2
			{-0.3125, -0.375, -0.3125, 0.3125, -0.25, 0.3125}, -- NodeBox3
			{-0.25, -0.5, -0.25, 0.25, -0.375, 0.25}, -- NodeBox4
			{-0.3125, 0.3125, -0.3125, 0.3125, 0.375, 0.3125}, -- NodeBox5
		}
	},
	S("Clay Water Pot with Salt Water"),
	{dig_immediate = 2})


--clay pot with freshwater
liquid_store.register_stored_liquid(
	"nodes_nature:freshwater_source",
	"tech:clay_water_pot_freshwater",
	"tech:clay_water_pot",
	{
		"tech_water_pot_water.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png",
		"tech_pottery.png"
	},
	{
		type = "fixed",
		fixed = {
			{-0.25, 0.375, -0.25, 0.25, 0.5, 0.25}, -- NodeBox1
			{-0.375, -0.25, -0.375, 0.375, 0.3125, 0.375}, -- NodeBox2
			{-0.3125, -0.375, -0.3125, 0.3125, -0.25, 0.3125}, -- NodeBox3
			{-0.25, -0.5, -0.25, 0.25, -0.375, 0.25}, -- NodeBox4
			{-0.3125, 0.3125, -0.3125, 0.3125, 0.375, 0.3125}, -- NodeBox5
		}
	},
	S("Clay Water Pot with Freshwater"),
	{dig_immediate = 2})


--make freshwater Pot drinkable on click
minetest.override_item("tech:clay_water_pot_freshwater",{
	on_rightclick = function(pos, node, clicker, itemstack, pointed_thing)
		local meta = clicker:get_meta()
		local thirst = meta:get_int("thirst")
		--only drink if thirsty
		if thirst < 100 then

			local water = 100 --you're skulling a whole bucket
			thirst = thirst + water
			if thirst > 100 then
				thirst = 100
			end

			--could add disease risk, but different sources have different risks
			--e.g. rain vs mud puddle

			meta:set_int("thirst", thirst)
			minimal.switch_node(pos, {name = "tech:clay_water_pot"})
			minetest.sound_play("nodes_nature_slurp",	{pos = pos, max_hear_distance = 3, gain = 0.25})
		end
	end
})
